UI/EncyclopediaDetailPanel.cpp

```diff
@@ -939,34 +939,56 @@ void EncyclopediaDetailPanel::HandleLinkDoubleClick(const std::string& link_type
     }
 }
 
+namespace {
+    std::unordered_map<std::string, std::pair<std::string, std::string>>
+        GetSubDirs(const std::string& dir_name, int depth = 0)
+    {
+        depth++;
+        std::unordered_map<std::string, std::pair<std::string, std::string>> retval;
+        // safety check to pre-empt potential infinite loop
+        if (depth > 50) {
+            WarnLogger() << "Exceeded recursive limit with lookup for pedia"
+                         << " category " << dir_name;
+            return retval;
+        }
+
+        std::unordered_map<std::string, std::pair<std::string, std::string>> sub_dirs;
+        std::multimap<std::string, std::pair<std::string, std::string>> sorted_entries;
+        GetSortedPediaDirEntires(dir_name, sorted_entries);
+        for (auto&& entry : sorted_entries) {
+            // explicitly exclude textures
+            if (entry.second.second == "ENC_TEXTURES")
+                continue;
+            sub_dirs.emplace(entry.second.second,
```

Comment:
Not in this case, braced init list has no type.
http://stackoverflow.com/questions/24229580/why-variadic-function-cant-eat-the-list-initialization-argument-in-c11